pkgs/nix-tools/darwin-rebuild.sh

```diff
@@ -16,6 +16,14 @@ showSyntax() {
   exit 1
 }
 
+sudo() {
+  if command sudo --help | grep -- --preserve-env= >/dev/null; then
+    command sudo -H --preserve-env=PATH "$@"
+  else
+    command sudo -H -- sh -c 'PATH='"'$PATH'"' "$@"' -- "$@"
```

Comment:
Not a big fan of how brittle that command looks, I would personally much rather not have `$PATH` passed, as it was only implemented for #798 and I'd consider it if more people ran into the same issue

```suggestion
    command sudo -H "$@"
```